### DISCUSSION TITLE
layui+vite进行项目工程化

### DISCUSSION THREAD

**Discussion by elylon1101** [layui/layui#1589] [分享]
使用layui+vite进行项目工程化
详情可以参考此项目[https://github.com/elylon1101/layui-vite](https://github.com/elylon1101/layui-vite)
主要功能点：
1. 使用es模块标准
2. 支持npm
3. 类似小程序的页面组成方式（xxx.css、xxx.js、xxx.html三个文件组成一个页面）
4. 封装了接口请求方法（[util/Request.js](https://github.com/elylon1101/layui-vite/blob/main/src/util/Request.js)内部使用fetch封装）
5. 无侵入优化layui table模块，支持请求拦截和url简写，直接在请求接口时处理登录过期等问题（[util/GlobalUtils.js](https://github.com/elylon1101/layui-vite/blob/main/src/util/GlobalUtils.js)，使用Proxy代理table的done方法）
6. 支持加载基于layui标准的外部模块，一般用于使用社区提供的其他模块
7. 内部实现了一个简单的响应式模块

欢迎大家一起完善

**Comment by elylon1101** [comment]
table模块的优化还支持请求接口时自动传递鉴权请求头

**Comment by cvpv** [comment]
都使用vite 了，直接用vue,react也是很好选择。

使用layui很大可能是要求必须基于原生开发，不使用任何打包工具，以便进行模板引擎渲染（php、jsp、thymeleaf）
